### PR TITLE
fix(diamond): remove token and add more roles to the manager

### LINKED ISSUE #577: Extract dollar form diamond and improve access control granularity for all tokens
we should extract the dollar ERC20 from the diamond and the access control from the diamond should be injected in the constructor
- add the dollar address inside the manager facet
- adapt all the facets and tests that reference it 
- create new burn and mint  roles for each of the ubiquity token

the dollar should be behind a UUPS proxy in order to be able to upgrade it ( we should have the tests improved to assess the upgradability)

### PR DESCRIPTION
**Pull request by zgorizzo69**
Resolves #577

### CODE REVIEW

File: packages/contracts/src/diamond/libraries/LibCreditNFTManager.sol [hunk 1]
```diff
@@ -153,10 +152,10 @@ library LibCreditNftManager {
         );
 
         // we burn user's dollars.
-        LibDollar.burn(msg.sender, amount);
+        IERC20Ubiquity(s.dollarTokenAddress).burnFrom(msg.sender, amount);
         // mint Credit
         UbiquityCreditToken creditToken = UbiquityCreditToken(
-            LibAppStorage.appStorage().creditTokenAddress
+            s.creditTokenAddress
```
Comment (0x4007) [inline packages/contracts/src/diamond/libraries/LibCreditNFTManager.sol]: Not a fan of one letter identifiers. Why not just call it storage or store

File: packages/contracts/src/diamond/token/ERC20UbiquityForDiamond.sol [hunk 2]
```diff
@@ -28,35 +27,19 @@ contract ERC20UbiquityForDiamond is
...
-        );
-        _;
-    }
-
+    // modifiers
     modifier onlyPauser() {
         require(
             accessCtrl.hasRole(PAUSER_ROLE, msg.sender),
-            "Governance token: not pauser"
+            "ERC20: not pauser"
```
Comment (0x4007) [inline packages/contracts/src/diamond/token/ERC20UbiquityForDiamond.sol]:
```suggestion
            "ERC20Ubiquity: not pauser"
```

File: packages/contracts/src/diamond/libraries/LibCreditNFTManager.sol (hunk 1 above)
Comment (zgorizzo69) [reply packages/contracts/src/diamond/libraries/LibCreditNFTManager.sol]: storage can't be used as a variable name as it is a keyword in solidity  
will rename to store then

File: packages/contracts/src/diamond/token/ERC20UbiquityForDiamond.sol (hunk 2 above)
Comment (zgorizzo69) [reply packages/contracts/src/diamond/token/ERC20UbiquityForDiamond.sol]: ok done

**Comment by 0x4007** [comment]
Looks good to me!

**Pull request merged**